src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -950,6 +950,7 @@ private static void Parse(
             ref StackRowStack stack)
         {
             bool inArray = false;
+            int numberOfProperties = 0;
```

Comment:
For my education, how do these variables work in cases where you have objects nested within other objects? I'm assuming this works because you're just copying the approach for arrays that presumably work when you have arrays within arrays.

Reply:
Exactly as you said. It works the same for arrays. 
When a new object is encountered (`StartObject`), the current variable value is stored on the stack before it is set to zero, so that we can continue when we reach the corresponding `EndObject` again.
I also added (and extended existing) unit tests to test nested object behaviour.

Reply:
Is it possible that you can just reuse the same local for both objects and arrays in that case? I'm guessing these would never overlap for the same frame.

Reply:
It should be possbile to reuse the same local variable. If you want, I can do the changes later.